File: client/src/lib/db/trigger.ts
import {nc, prisma} from "@/lib/singleton";;

export async function getTriggerByIdAndOrganization(organizationId: string, userId: string, scriptId: string) {
  return prisma.trigger.findFirst({
    select: {
      id: true,
      name: true,
      code: true,
      enable: true,
      channel: true,
      conditions: {
        select: {
          id: true,
          name: true,
          type: true,
          code: true,
          timeout: true,
          enable: true
        }
      }
    },
    where: {
      id: scriptId,
      organizationId,
      organization: {
        OR: [
          {
            creatorId: userId,
          },
          {
            users: {
              some: {
                userId
              }
            }
          }
        ]
      }
    }
  })
}

File: backend/internal/model/trigger.go
package model

type Trigger struct {
	Id             string `json:"id"`
	OrganizationId string `json:"organization"`
	Code           string `json:"code"`
	Name           string `json:"name"`
	Enable         bool   `json:"enable"`
	Channel        string `json:"channel"`
}

type TriggerConditionExpand struct {
	Trigger Trigger `json:"trigger"`
}

type TriggerCondition struct {
	Id      string                 `json:"id"`
	Code    string                 `json:"code"`
	Name    string                 `json:"name"`
	Enable  bool                   `json:"enable"`
	Type    string                 `json:"type"`
	Timeout int                    `json:"timeout"`
	Expand  TriggerConditionExpand `json:"expand"`
}
